_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,273 @@
+                   _______|_________
+                  /                \
+?resourceVersion=rel:latest-version
+                  \_/ \____________/
+                   |        |
+         version-negotiator |
+                     version-argument
+```
+
+## Server Responsibilities
+
+<a id="bad-version-negotiator"></a>A server **MUST** respond with `400 Bad Request` if a version negotiator is not
+supported. In this case, an error object that includes a `type` link to
+`https://jsonapi.org/profiles/drupal/resource-versioning/#bad-version-negotiator`
+**MUST** be included in the response document.
+   
+<a id="bad-version-argument"></a>If a server cannot process the given version argument for the given negotiation
+mechanism, it **MUST** respond with a `400 Bad Request`. In this case, an error
+object that includes a `type` link to
+`https://jsonapi.org/profiles/drupal/resource-versioning/#bad-version-argument`
+**MUST** be included in the response document.
+
+If a server is able to process the version argument but an appropriate version
+cannot be located, the server **MUST** respond with a `404 Not Found`.
+
+# Links
+
+When a server processes a request with a `resourceVersion` query parameter and
+a `self` link is provided for a top-level links object, the link's `href`
+**MUST** include the `resourceVersion` query parameter with the same version
+identifier that was requested.
+
+When a server processes a request with a `resourceVersion` query parameter
+all resource object `self` links **SHOULD** contain a `resourceVersion` query
+parameter which identifies the specific revision represented by that resource
+object.
+
+For example, in the following response document the `self` links are not the
+same:
+
+```json
+{
+  "data": {
+    "type": "article",
+    "id": 1,
+    "links": {
+      "self": "/article/1?resourceVersion=id:42"
+    }
+  },
+  "links": {
+    "self": "/article/1?resourceVersion=rel:latest-version"
+  }
+}
+```
+
+A server **MAY** provide a `version-history` link in a resource object's links
+object if a [version history](version-history) endpoint is available for the context resource
+object.
+
+A server **MAY** provide the following resource object links so that a client may
+navigate a resource object's version history:
+
+  - `latest-version`: links to the latest version of the context resource
+    object.
+  - `working-copy`: links to the working copy of the context resource object.
+  - `predecessor-version`: links to the version which immediately preceded the
+    context resource object.
+  - `successor-version`: links to the version which immediately succeeded the
+    context resource object.
+  - `prior-working-copy`: links to the working copy which immediately preceded
+    the context resource object.
+  - `subsequent-working-copy`: links to the working copy which immediately
+    preceded the context resource object.
+
+A server **MAY** provide an array of any of these links to support branching.
```

Comment:
@gabesullice I'd be happy to just define these links in the base spec as a temporary workaround. Note also that the base spec's current approach is that, if an array of links is allowed, an array should be required (like we did for `profile`) to not further proliferate the number of links cases that clients have to handle. 

So, the base spec language would go under the [resource links](https://jsonapi.org/format/#document-resource-object-links) section, and could say something like:

> If present, this object **MAY** also contain an array of links as the value for any of the following keys: `latest-version`, `working-copy`, `predecessor-version`, `successor-version`, [etc]. Links at any of these keys point to resources that are related to the resource object according to the IANA link relation corresponding to their key name.

Reply:
That'd be great :)

Reply:
bump.

Reply:
@ethanresnick Rather than calling out those 4 specific link relations, perhaps it'd be better to generalize that to something like:
> Links with a key that matches a [registered link relation type](
https://www.iana.org/assignments/link-relations/link-relations.xhtml) which allows multiple values are allowed to have a value that is not a link, but an array of links.

Reply:
While I agree that calling out specific link relations "smells" a bit, I would not like to see the concept of links object keys as link relations furthered in the spec. Links can have more than one link relation and this pattern makes that more challenging to fix in later iterations of the spec.